src/nf/nf_random_submodule.f90

```diff
@@ -32,4 +32,22 @@ module function randn_4d(i, j, k, l) result(x)
     x = sqrt(-2 * log(u)) * cos(2 * pi * v)
   end function randn_4d
 
+  module impure elemental subroutine randn_ele(x)
+    real, intent(out) :: x
+    real :: u(2), r
+    real, save :: z
+    logical, save :: have_z = .false.
+    if (have_z) then
+        x = z
+        have_z = .false.
+    else
+        call random_number(u)
+        u(1) = 1.0 - u(1)
```

Comment:
Should it be done for the whole vector, or only for the first element? In other words, might u(2) be equal to 0 (I don't know the answer)?

Reply:
I thought it was fine because `u(2)` doesn't go into `log()`.